lib/queue_bus/config.rb

```diff
@@ -6,16 +6,46 @@
 module QueueBus
   # This class contains all the configuration for a running queue bus application.
   class Config
-    attr_accessor :default_queue, :local_mode, :hostname, :incoming_queue, :logger
+    attr_accessor :default_queue, :hostname, :incoming_queue, :logger
 
     attr_reader :worker_middleware_stack
+    attr_writer :local_mode
 
     def initialize
       @worker_middleware_stack = QueueBus::Middleware::Stack.new
       @incoming_queue = 'bus_incoming'
       @hostname = Socket.gethostname
     end
 
+    # A wrapper that is always "truthy" but can contain an inner value. This is useful for
+    # checking that a thread local variable is set to a value, even if that value happens to
+    # be nil. This is important because setting a thread local value to nil will cause it to
+    # be deleted.
+    Wrap = Struct.new(:value)
+
+    LOCAL_MODE_VAR = :queue_bus_local_mode
+
+    # Returns the current local mode of QueueBus
+    def local_mode
```

Comment:
As implemented, this method says "if you've ever used `with_local_mode` then return it's current value" and otherwise, return the "all_thread" global default.

I think that might lead to a problem in long running threads. I think we'd rather have "if not currently in `with_local_mode`, return the "all_thread" global default.

One way to do that might actually using the `nil` behavior above so that `Thread.current.thread_variable?` return false when not in the middle of the `with_local_mode` block.

Reply:
Or am I wrong, because the `previous` before is `nil` in that case it gets set back to `nil` in the `ensure` and it works as I hoped. That seems possible. Still leaving these comments just to confirm.

Reply:
Yup, that's why I introduced the `Wrap` class so that truthiness is more deterministic.

Reply:
Word. Thanks.